### DISCUSSION TITLE
`@storybook/addon-docs` Props reading from declaration file

### DISCUSSION THREAD

**Discussion by ashalfarhan** [storybookjs/storybook#16721] [Help] [answered]
Hi, i'm figuring out on how to make `Props` component to read the prop type from compiled module.
Because currently i'm creating a design system and separate the desktop version and mobile version to dedicated package.
And then creating another package for documentation for both package in this one package with storybook.

My question is, is it possible to make `Props` component from storybook addon-docs to read declaration file (d.ts) of the respective component?, because as far as i know, `Props` component can only read prop type from `.tsx` file.

**Comment by shilman** [comment, marked as answer]
Storybook currently uses `react-docgen-typescript` for extracting type information from components: https://storybook.js.org/docs/react/configure/typescript#mainjs-configuration

However our `argTypes` data structure is not too complicated. If you can't get `react-docgen-typescript` to work, perhaps you can figure out how to convert from `.d.ts` files to `argTypes` directly yourself.

**Reply by ashalfarhan** [reply]
And also, i've been following that guide to figure out the `propFilter` function, but in below snippet, the `propFilter` is not called, am i missed something?

```ts
typescript: {
    reactDocgen: 'react-docgen-typescript',
    reactDocgenTypescriptOptions: {
      include: ["../../../node_modules/@my-lib/**/*.d.ts"],
      // This is not called
      propFilter: (prop) =>  {console.log("prop:", prop.parent); process.exit(1)}
    },
 },
```

**Reply by slimshreydy** [reply]
Hi @ashalfarhan, did you ever figure this out? I am running into the same issue; trying to pull docs from the .d.ts file corresponding to a component.
